src/pcp/types.rs

```diff
@@ -356,59 +369,83 @@ mod tests {
                 expect_valid: false,
                 expected_proof_len: 16,
             },
-        );
+        )
+        .unwrap();
     }
 
-    // TODO(cjpatton) Have this return an error and have the caller assert success (or failure).
-    fn pcp_validity_test<V: Value>(input: &V, t: &ValidityTestCase) {
+    fn pcp_validity_test<V: Value>(input: &V, t: &ValidityTestCase) -> Result<(), PcpError> {
```

Comment:
Having a `cfg(test)`-only variant on `PcpError` is totally fine, but another option would be to have this return `Result<(), String>`, and then do `return Err(format!("whatever {}", something));`. That is, unless you also need to propagate other variants of `PcpError` from this function.

Reply:
Since the same function may propagate a `PcpError`, adding the variant seemed like the simplest option. Otherwise we'd have to convert `PcPError`s to strings explicitly. (This wouldn't be so bad...)

Reply:
Since you need to handle other `PcpError` variants, let's keep what you have and which already works.